src/URI.js

```diff
@@ -524,6 +524,8 @@
     // what's left must be the path
     parts.path = string;
 
+    URI.basicValidation(parts);
```

Comment:
The validation should probably be done in [`URI.parseHost()`](https://github.com/medialize/URI.js/blob/gh-pages/src/URI.js#L530)? maybe [`URI#hostname()`'s validation](https://github.com/medialize/URI.js/blob/gh-pages/src/URI.js#L1307) can be moved there, too?